### PR TITLE
Update security-secrets-setup README.md

### PR DESCRIPTION
**Pull request by jim-wang-intel**
Add instructions to run security-secrets-setup various subcommands in stand-alone executable

Fix typo as found in issue #1815

Signed-off-by: Jim Wang <yutsung.jim.wang@intel.com>

### CODE REVIEW

File: cmd/security-secrets-setup/README.md [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
 make docker_security_secrets_setup
 ```
 
 It should create a docker image with the name `edgexfoundry/docker-edgex-vault:<version>-dev` if sucessfully built.
 
-# Run
+## Docker Run
 
 To see what the TLS materials are generated inside the docker container, 
 one can run the built docker image above (assuming the version number is `1.1.0`) as:
```
Comment (bnevis-i) [inline cmd/security-secrets-setup/README.md]: Docker Run section is still showing the legacy output - should add some text to note that.

Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: Right, that is the current way it is in docker image...this will get changed when the docker image split-ie. after the other issue #1716 is addressed.

**Comment by jim-wang-intel** [comment]
cc: @eno-intel

File: cmd/security-secrets-setup/README.md [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+Clean up `/run/edgex/secrets` once done verification if necessary.
+
+## Executable Run [cache] subcommand
+
+Before cache subcommand is issued, make sure cache dir (in this example, it is */tmp/pki/cache*) is non-existing or empty.
+
+To run the **cache** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run 
+
+```sh
+mkdir -p /tmp/pki/cache && export PKI_CACHE=/tmp/pki/cache && sudo -E ./security-secrets-setup cache
```
[the remark this answers is not in the document]
Comment (anonymouse64) [reply cmd/security-secrets-setup/README.md]: finally these kinds of commands read better in a shell session with one command per line as in:

```suggestion
$ mkdir -p /tmp/pki/cache
$ export PKI_CACHE=/tmp/pki/cache
$ sudo -E ./security-secrets-setup cache
```

File: cmd/security-secrets-setup/README.md [hunk 3]
```diff
@@ -1,16 +1,88 @@
...
+```
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
+
+## Executable Run [import] subcommand
+
+To run the **import** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run 
+
+```sh
+mkdir -p /tmp/pki/cache && export PKI_CACHE=/tmp/pki/cache && sudo -E ./security-secrets-setup import
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]: see above comment about having one command per line

File: cmd/security-secrets-setup/README.md [hunk 4]
```diff
@@ -1,16 +1,88 @@
...
+Only in /run/edgex/secrets/ca: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-kong: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-vault: .security-secrets-setup.complete
+```
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
+
+## Executable Run [import] subcommand
+
+To run the **import** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]:
```suggestion
To use the **import** subcommand, from the same `cmd/security-secrets-setup` directory run 
```

File: cmd/security-secrets-setup/README.md [hunk 5]
```diff
@@ -1,16 +1,88 @@
...
+
+## Executable Run [import] subcommand
+
+To run the **import** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run 
+
+```sh
+mkdir -p /tmp/pki/cache && export PKI_CACHE=/tmp/pki/cache && sudo -E ./security-secrets-setup import
+```
+
+There are two cases can happen depends on the status of cache directory:
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]:
```suggestion
The **import** subcommand operates differently depending on the status of cache directory:
```

File: cmd/security-secrets-setup/README.md [hunk 6]
```diff
@@ -1,16 +1,88 @@
...
+
+1. Cache dir */tmp/pki/cache* is empty:
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]: Since there's only 2 possibilities, I think it's simpler to just use plain English and not enumerate them in a number list, like so:

```suggestion
If the cache dir `/tmp/pki/cache` is empty, then the command blocks and waits for ....
```

(also note this now out-of-date per our Slack conversation regarding the blocking...)

File: cmd/security-secrets-setup/README.md [hunk 7]
```diff
@@ -1,16 +1,88 @@
...
+
+    In this case, one should expect to see the execution is blocking and waiting for TLS assets to be drop into cache directory before they can be deployed into directory */run/edgex/secrets/*.
+
+2. Cache dir */tmp/pki/cache* has already loaded or cached with generated TLS assets before run import subcommand:
+
+    One should see the deploying TLS assets happening right away from cached directory */tmp/pki/cache* into deployed directory */run/edgex/secrets*.  One can verify that there are Vault and Kong TLS assets deployed from cache directory */tmp/pki/cache* onto the directory */run/edgex/secrets* via `ls` command.
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]: Since this paragraph is just about verifying the files exist I think we should drop it. We already mention above that the files are copied for deployment, so I don't think it's necessary to explain again that files are deployed.

```suggestion
```

File: cmd/security-secrets-setup/README.md [hunk 8]
```diff
@@ -1,16 +1,88 @@
...
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]: see above comment about cleaning things up

File: cmd/security-secrets-setup/README.md [hunk 9]
```diff
@@ -1,16 +1,88 @@
...
+
+```sh
+sudo diff -r /tmp/pki/cache/ /run/edgex/secrets/
+
+Only in /run/edgex/secrets/ca: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-kong: .security-secrets-setup.complete
+Only in /run/edgex/secrets/edgex-vault: .security-secrets-setup.complete
+```
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]: see above comment about clean up comments

File: cmd/security-secrets-setup/README.md [hunk 10]
```diff
@@ -1,16 +1,88 @@
...
+
+2. Cache dir */tmp/pki/cache* has already loaded or cached with generated TLS assets before run import subcommand:
+
+    One should see the deploying TLS assets happening right away from cached directory */tmp/pki/cache* into deployed directory */run/edgex/secrets*.  One can verify that there are Vault and Kong TLS assets deployed from cache directory */tmp/pki/cache* onto the directory */run/edgex/secrets* via `ls` command.
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
+
+## Docker Build
+
+Use Makefile in the base directory of this repository to build the docker image of security-secrets-setup:
```
Comment (anonymouse64) [inline cmd/security-secrets-setup/README.md]:
```suggestion
Use the Makefile in the root directory of the repository to build the docker container image for security-secrets-setup:
```

File: cmd/security-secrets-setup/README.md [hunk 11]
```diff
@@ -1,16 +1,88 @@
...
+```
+
+It should create a binary executable under the directory *cmd/security-secrets-setup/* if successfully built.
+
+## Executable Run [generate] subcommand
+
+To run the **generate** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run 
+
+```sh
+sudo -E ./security-secrets-setup generate
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: it needs root privilege to do things

File: cmd/security-secrets-setup/README.md [hunk 12]
```diff
@@ -1,16 +1,88 @@
...
+```
+
+Verify that it runs successfully with generating both Vault and Kong's TLS assets under `/run/edgex/secrets/` directory.
+
+Clean up `/run/edgex/secrets` once done verification if necessary.
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: ok to remove this

File: cmd/security-secrets-setup/README.md [hunk 13]
```diff
@@ -60,11 +60,11 @@ To run the **import** subcommand in linux system, change directory to *cmd/secur
 mkdir -p /tmp/pki/cache && export PKI_CACHE=/tmp/pki/cache && sudo -E ./security-secrets-setup import
 ```
 
-There are two cases can happen depends on the status of cache directory:
+There are two cases can happen depending on the status of cache directory:
 
 1. Cache dir */tmp/pki/cache* is empty:
 
-    In this case, one should expect to see the execution is blocking and waiting for TLS assets to be drop into cache directory before they can be deployed into directory */run/edgex/secrets/*.
+    In this case, import just errors out: the TLS assets is required to be in the cache directory before they can be deployed into directory */run/edgex/secrets/*.
```
Comment (bnevis-i) [inline cmd/security-secrets-setup/README.md]: The intent here was that if the assets are being copied from an image into a mounted docker volume, that the container would wait for this to finish.  Does docker guarantee that this will happen before the container starts executing?

File: cmd/security-secrets-setup/README.md (hunk 6 above)
Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: i think list reads better, I agree with you on not using the number list but I am ok with changing it to a bullet list not numbered since number list implies the sequence and priority

File: cmd/security-secrets-setup/README.md (hunk 7 above)
Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: i think for the sake of completeness, this should retain

File: cmd/security-secrets-setup/README.md [hunk 14]
```diff
@@ -1,16 +1,88 @@
...
+
+## Executable Run [generate] subcommand
+
+To run the **generate** subcommand in linux system, change directory to *cmd/security-secrets-setup/* and then run 
+
+```sh
+sudo -E ./security-secrets-setup generate
+```
+
+Verify that it runs successfully with generating both Vault and Kong's TLS assets under `/run/edgex/secrets/` directory.
```
[the remark this answers is not in the document]
Comment (bnevis-i) [reply cmd/security-secrets-setup/README.md]: Will also be able to fix some of this when support for `configuration.toml` is added, as /run is really for docker, and it can be in the `res/docker/configuraiton.toml` and the standard `configuration.toml` can just write something to `/tmp/somewhere`

**Comment by bnevis-i** [comment]
Any chance that we could refer to the [documentation page](https://github.com/edgexfoundry/edgex-docs/blob/master/security/security-secrets-setup.1.rst) for some of the details rather than duplicate them in the README?

**Comment by jim-wang-intel** [comment]
@bnevis-i I can put a link to the documentation page at the end of this notes as reference.  This document is more about how to build and run for testing purpose; not sure what the duplicates this one bring?

**Comment by bnevis-i** [comment]
> @bnevis-i I can put a link to the documentation page at the end of this notes as reference. This document is more about how to build and run for testing purpose; not sure what the duplicates this one bring?

Like for instance, when documenting what PKI_CACHE_DIR does, I'm suggesting to just point to the main docs and say "this sets CacheDir via an environment variable" without reproducing all the gory details in the README.

File: cmd/security-secrets-setup/README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: yea, will need to modify this doc. again when that feature is in.

File: cmd/security-secrets-setup/README.md [hunk 15]
```diff
@@ -2,15 +2,87 @@
...
+- Cache dir `/tmp/pki/cache` has already loaded or cached with generated TLS assets before run **import** subcommand:
+
+    One should see the deploying TLS assets from the cached directory `/tmp/pki/cache` into the deployed directory `/run/edgex/secrets`.  One can verify that there are Vault and Kong TLS assets deployed from cache directory `/tmp/pki/cache` onto the directory `/run/edgex/secrets` via `ls` command.
+
+## Docker Build
+
+Use the Makefile in the root directory of the repository to build the docker container image for `security-secrets-setup`:
 
 ```sh
 make docker_security_secrets_setup
```
Comment (Walt-Intel) [inline cmd/security-secrets-setup/README.md]: When I get to the 'make docker_security_secrets_setup' step I see this error message:

```$ make docker_security_secrets_setup
docker build \
	-f cmd/security-secrets-setup/Dockerfile \
	--label "git_sha=21750bd69222753f69a7a217e480b07ca6bb6dd6" \
	-t edgexfoundry/docker-edgex-vault:21750bd69222753f69a7a217e480b07ca6bb6dd6 \
	-t edgexfoundry/docker-edgex-vault:1.1.0-dev \
	.
error checking context: 'can't stat '/home/tpm/go-jim/edgex-go/cmd/security-secrets-setup/logs/security-secrets-setup.log''.
Makefile:174: recipe for target 'docker_security_secrets_setup' failed
make: *** [docker_security_secrets_setup] Error 1
```

Looking at the directory structure of the secrets_setup I see the log file are owned by 'root':
```$ ls -la /home/tpm/go-jim/edgex-go/cmd/security-secrets-setup
-rwxrwxr-x  1 tpm  tpm      3012 Oct  2 10:28 entrypoint.sh
drwxr--r--  2 root root     4096 Oct  2 10:30 logs
-rw-rw-r--  1 tpm  tpm      4514 Oct  2 10:28 main.go
```

Comment (jim-wang-intel) [reply cmd/security-secrets-setup/README.md]: yeap, please go to the `root` directory- ie. `edgex-go/` to run the docker build: `make docker_security_secrets_setup` . that way it should be fine. I'll mention that in the document about the root directory you should be run the make docker

**Comment by jim-wang-intel** [comment]
Rebase

**Comment by anonymouse64** [comment]
I don't have time to re-review this, please continue without me
